services/QuillLMS/engines/evidence/spec/workers/evidence/research/gen_ai/run_trial_evaluation_worker_spec.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'sidekiq/testing'
+
+RSpec.describe Evidence::Research::GenAI::RunTrialEvaluationWorker, type: :worker do
+  let(:trial) { create(:evidence_research_gen_ai_trial) }
+  let(:version) { described_class::VERSION }
+  let!(:suboptimal_example1) { create(:evidence_research_gen_ai_llm_example, :suboptimal, trial:) }
+  let!(:suboptimal_example2) { create(:evidence_research_gen_ai_llm_example, :suboptimal, trial:) }
+  let!(:optimal_example1) { create(:evidence_research_gen_ai_llm_example, :optimal, trial:) }
+
+  describe '#perform' do
+    subject { described_class.new.perform(trial.id) }
+
+    it 'enqueues feedback evaluation jobs for suboptimal examples' do
+      expect(Evidence::Research::GenAI::RunFeedbackEvaluationWorker).to receive(:perform_async)
+        .with(trial.id, version, suboptimal_example1.id)
+      expect(Evidence::Research::GenAI::RunFeedbackEvaluationWorker).to receive(:perform_async)
+        .with(trial.id, version, suboptimal_example2.id)
+
+      subject
+    end
+  end
+
+  describe '#on_complete' do
+    let(:options) { { 'trial_id' => trial.id } }
+
+    subject { described_class.new.on_complete(nil, options) }
+
+    it 'sets evaluator counts' do
+      expect_any_instance_of(Evidence::Research::GenAI::Trial).to receive(:set_evaluator_counts)
```

Comment:
RSpec [documentatation](https://relishapp.com/rspec/rspec-mocks/docs/working-with-legacy-code/any-instance) recommends not using `expect_any_instance_of/allow_any_instance_of`

Instead you could stub find by trial.id:
```
describe '#on_complete' do
  let(:options) { { 'trial_id' => trial.id } }
  let(:trial_instance) { instance_double(Evidence::Research::GenAI::Trial) }

  before do
    allow(Evidence::Research::GenAI::Trial).to receive(:find).with(trial.id).and_return(trial_instance)
  end

  subject { described_class.new.on_complete(nil, options) }

  it 'sets evaluator counts' do
    expect(trial_instance).to receive(:set_evaluator_counts)
    subject
  end
end
```

Reply:
Will update!